src/test/java/com/ifactory/press/db/solr/search/SafariBlockJoinTest.java

```diff
@@ -50,7 +54,28 @@ public void testChildMatch () throws Exception {
     SolrDocument doc = solr.query(query).getResults().get(0);
     assertEquals ("/doc/14", doc.get("uri").toString());
   }
-  
+
+  @Test
+  public void testChildMatchExplanation () throws Exception {
+    SolrQuery query = new SolrQuery("{!scoring_parent which=type_s:parent} text_t:M");
+    query.setShowDebugInfo(true);
+    QueryResponse resp = solr.query(query);
+    Map<String, String> explainMap = resp.getExplainMap();
+
+    // expect to get back docs 14, 28, 42, 56, 70, 84, 98
+    HashSet<String> expectedKeys = new HashSet<String>(
```

Comment:
generally preferable to declare at the least restrictive level possible (so `Set`, not `HashSet`)...

 ```
Set<String> expectedKeys = new HashSet<String>(
``` 

...and as i mentioned in chat, the second parameterization is not need in 7+, but is fine to leave just in case

Reply:
Ah, I saw that `Set` was just an interface and for some reason I thought you couldn't use interfaces in declarations like that. I'll give that a try.